pinot-core/src/main/java/org/apache/pinot/core/plan/GroupByPlanNode.java

```diff
@@ -75,35 +70,10 @@ private GroupByOperator buildNonFilteredGroupByPlan() {
 
     FilterPlanNode filterPlanNode = new FilterPlanNode(_indexSegment, _queryContext);
     BaseFilterOperator filterOperator = filterPlanNode.run();
-
-    // Use star-tree to solve the query if possible
-    List<StarTreeV2> starTrees = _indexSegment.getStarTrees();
-    if (starTrees != null && !_queryContext.isSkipStarTree()) {
```

Comment:
Note: this PR would introduce a behaviour change (bugfix?) where `AggregationPlanNode` included an additional predicate to check if null handling is enabled before allowing use of StarTree index to solve the query.  This PR consolidates this logic to one method within OperatorUtils.java, where the null handling predicate is kept such that both GroupByPlanNode and AggregationPlanNode will ensure null handling is disabled before allowing use of StarTree index.

> if (starTrees != null && !_queryContext.isSkipStarTree() && !_queryContext.isNullHandlingEnabled()) {